### ISSUE TITLE
feat: token-level loss normalization (fixed constant divisor)

### ISSUE THREAD

**Issue by teilomillet** [teilomillet/textpolicy#8]
## Parent Issue
Part of #5 (GRPO optimizations for reasoning emergence)

## Problem

Current `policy_loss()` uses:
```python
loss = -mx.mean(mx.minimum(surr1, surr2))
```

This averages over all tokens, which can cause:
- Longer responses have disproportionately lower per-token contribution
- Creates implicit length bias (short responses have outsized influence)
- Incorrect responses tend to be longer, amplifying this bias

## Research Evidence

Dr. GRPO (Understanding R1-Zero-Like Training) recommends:
> "Divide sequence loss by a constant rather than token count to eliminate length bias that artificially inflates incorrect response lengths."

**Source:** [Understanding R1-Zero-Like Training: A Critical Perspective](https://arxiv.org/abs/2503.20783)

## Proposed Change

```python
def policy_loss_token_normalized(
    old_logprobs: mx.array,
    new_logprobs: mx.array,
    advantages: mx.array,
    clip_ratio_low: float = 0.2,
    clip_ratio_high: float = 0.28,
    normalize_constant: int = 1024  # Fixed constant instead of token count
) -> mx.array:
    """
    Policy loss with fixed-constant normalization.
    
    Uses sum/constant instead of mean to eliminate length bias.
    All sequences contribute equally regardless of length.
    """
    ratio = mx.exp(new_logprobs - old_logprobs)
    clipped_ratio = mx.clip(ratio, 1 - clip_ratio_low, 1 + clip_ratio_high)
    surr1 = ratio * advantages
    surr2 = clipped_ratio * advantages
    
    # Sum and divide by constant, not by token count
    loss = -mx.sum(mx.minimum(surr1, surr2)) / normalize_constant
    
    return loss
```

## Alternative: Batch-wise Normalization

Another option from the research:
```python
# Normalize by batch size instead of total tokens
loss = -mx.sum(mx.minimum(surr1, surr2)) / batch_size
```

## Impact

- Eliminates length bias in loss computation
- All sequences contribute equally regardless of length
- Aligns with Dr. GRPO recommendations
- Prevents incorrect (often longer) responses from dominating gradients

## Acceptance Criteria

- [ ] Add `normalize_constant` parameter to `policy_loss()`
- [ ] Default to reasonable constant (e.g., 1024) or batch_size
- [ ] Option to use original mean behavior for backward compatibility
- [ ] Unit tests comparing loss values for short vs long sequences
- [ ] Document the length bias issue and fix
